File: Gaming Website/rockpaperscissors.js
const gameDiv = document.getElementById("game");
const choicesDiv = document.getElementById("choices");
const playAgainButton = document.getElementById("play-again");

let playerChoice;

function simulateGame(playerChoice) {
  const choices = ["rock", "paper", "scissors"];
  const computerChoice = choices[Math.floor(Math.random() * 3)];

  const results = {
    rock: {
      rock: "Tie!",
      paper: "You lose!",
      scissors: "You win!",
    },
    paper: {
      rock: "You win!",
      paper: "Tie!",
      scissors: "You lose!",
    },
    scissors: {
      rock: "You lose!",
      paper: "You win!",
      scissors: "Tie!",
    },
  };

  const result = results[playerChoice][computerChoice];
  return { computer_choice: computerChoice, result };
}

function playGame() {
  // Display the available choices
  choicesDiv.style.display = "flex";

  // Add click event listeners to the choices buttons
  const choices = document.querySelectorAll(".choice");
  choices.forEach((choice) => {
    choice.addEventListener("click", () => {
      playerChoice = choice.id;
      choicesDiv.style.display = "none";

      // Simulate the game
      const data = simulateGame(playerChoice);

      const { computer_choice: computerChoice, result } = data;
      gameDiv.textContent = `You chose ${playerChoice}. Computer chose ${computerChoice}. ${result}`;
      playAgainButton.style.display = "block";
    });
  });
}

playAgainButton.addEventListener("click", () => {
  gameDiv.textContent = "";
  playAgainButton.style.display = "none";
  playGame();
});

playGame();

File: Gaming Website/rockpaperscissors.py
from flask import Flask, request, jsonify
import random

app = Flask(__name__)

@app.route('/play', methods=['POST'])
def play():
    try:
        choice = request.json['choice']
        if choice not in ['rock', 'paper', 'scissors']:
            raise ValueError("Invalid choice")
    except (KeyError, ValueError):
        return jsonify({'error': 'Invalid request'}), 400
    
    computer_choice = random.choice(['rock', 'paper', 'scissors'])
    result = determine_winner(choice, computer_choice)
    return jsonify({'computer_choice': computer_choice, 'result': result})

def determine_winner(player_choice, computer_choice):
    if player_choice == computer_choice:
        return "It's a tie!"
    elif (player_choice == "rock" and computer_choice == "scissors"
            or player_choice == "paper" and computer_choice == "rock"
            or player_choice == "scissors" and computer_choice == "paper"):
        return "You win!"
    else:
        return "Computer wins!"

if __name__ == '__main__':
    app.run()
